src/h5web/visualizations/nexus/NxImageContainer.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { ReactElement, useState, useContext } from 'react';
+import { range } from 'lodash-es';
+import { HDF5SimpleShape } from '../../providers/models';
+import { useDatasetValue } from '../shared/hooks';
+import { assertGroup } from '../../providers/utils';
+import DimensionMapper from '../../dataset-visualizer/mapper/DimensionMapper';
+import { DimensionMapping } from '../../dataset-visualizer/models';
+import { ProviderContext } from '../../providers/context';
+import { getSignalDataset } from './utils';
+import { VisContainerProps } from '../shared/models';
+import MappedHeatmapVis from '../heatmap/MappedHeatmapVis';
+
+function NxImageContainer(props: VisContainerProps): ReactElement {
+  const { entity } = props;
+  assertGroup(entity);
+
+  const { metadata } = useContext(ProviderContext);
+  const dataset = getSignalDataset(entity, metadata);
+
+  if (!dataset) {
+    throw new Error('Signal dataset not found');
+  }
```

Comment:
I look for the signal dataset again here (I already did once in `supportsEntity`), but repeating the call to `getSignalDataset` means we don't have any coupling between the code that checks support and the code that actually prepares the visualization. If an error is thrown here it is caught by the new error boundary.

Reply:
The error will never be thrown as an entity pointing to a non-valid `dataset` will be considered as not supported by the `NxImage`.

See my comment above in the `supportsEntity` function of `NxImage`

Reply:
Yep, yep, but the checks in the containers are important for two reasons:

- If the container ends up being used outside of h5web.
- For type checking purposes (we don't have to worry about the dataset being `undefined` in the rest of the container).

Reply:
Makes sense :+1: